api/src/error.rs

```diff
@@ -43,6 +43,9 @@ pub enum Error {
     #[error("Expected a valid bucket type, got: {}", bucket_type)]
     BadBucketType { bucket_type: String },
 
+    #[error("Expected valid json for entity defs, got: {}", entity_defs)]
+    BadEntityDefs { entity_defs: String },
```

Comment:
Github lost my comment...

```suggestion
    #[error(
        "Expected valid json for entity def. Got: '{}', which failed because: '{}'",
        entity_def,
        source
    )]
    BadEntityDef {
        entity_def: String,
        #[source]
        source: ::serde_json::Error,
    },
```

- nit: rename as it's just an error for a single entity def
- capture and display the underlying error message. Good for traceability, and more info to the user

```bash
cargo run -- create dataset org1/bar -e '{"foo":"bar"}'
error: Invalid value for '--entity-def <entity-defs>...': Expected valid json for entity def. Got: '{"foo":"bar"}', which failed because: 'missing field `name` at line 1 column 13'
```

Reply:
Excellent! Thanks!

Reply:
What does `#[source]` do actually?

Reply:
It a) does some magic allowing errors to be nested in this way IIRC, but most importantly it marks this error has having caused the parent one.

This allows methods like `source()` which returns the underlying error. `anyhow` uses this to do things like [`chain`](https://docs.rs/anyhow/1.0.38/anyhow/struct.Error.html#method.chain) which returns an iterator of all the errors that caused this one (kind of like a traceback)